### DISCUSSION TITLE
Access idToken and contents inside idToken

### DISCUSSION THREAD

**Discussion by ChristianMetzgerPinelta** [Budibase/budibase#9543] [Help]
Hi all,
we are testing BudiBase at the moment as a retool alternativ. We are using SSO over our own keykloak which is working just fine.
The other day I tried to access the idToken which is sent from the keykloak to use some stuff in it for another REST query.
I tried it the same way, it's shown when you look at the options of a data provider **{{ [user].[oauth2].[accessToken] }}**.
But it did not work any possible way i tied.

So i just wanted to ask if it is possible to access the idToken and possibly contents within it. Because ideally we want to use some roles, which are hidden inside the idToken to create custome user roles in budibase if that is possible and I want to use a id for another query which is hidden inside the idToken as well.

hopefully you can help me.

best regards

**Comment by melohagan** [comment]
Hey @ChristianMetzgerPinelta 

Would you be able to share some screenshots of what you are trying please?
